app/controllers/api/vms_controller.rb

```diff
@@ -351,7 +351,7 @@ def vm_event(vm, event_type, event_message, event_time)
     def retire_vm(vm, id, data)
       desc = "#{vm_ident(vm)} retiring"
       desc << " on #{data['date']}" if Hash(data)['date'].present?
-      retire_resource(:vms, id, data)
+      method(:retire_resource).super_method.call(:vms, id, data)
```

Comment:
I guess you *already* lose the polymorphism by having a `retire_vm` method, right?  It just feels like this is abusing the true polymorphic path.